src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,37 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationServiceException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int ALLOWED_USERS_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationServiceException("User is null");
+        }
+        User alreadyHaveSuchUser = storageDao.get(user.getLogin());
+        if (alreadyHaveSuchUser != null) {
```

Comment:
```suggestion
        if (storageDao.get(user.getLogin()) != null) {
```
don't need create new instances/ variables every time, you can directly use (storageDao.get(user.getLogin()) != null).